paasta_tools/cli/cmds/logs.py

```diff
@@ -744,6 +758,9 @@ def callback(
             clusters=clusters, components=components, callback=callback
         )
 
+        strip_cluster = True if len(clusters) < 2 else False
+        strip_instance = True if instances is not None else False
```

Comment:
I think we also want `len(instances) < 2` here? I don't think `instances` can be `None` here either because it's an `Iterable`, not an `Optional`.

Reply:
Seems like we don't need this logic anyway since we are only allowing one cluster and one instance?

Reply:
@kawaiwanyelp `instances` can actually be `None` in this case. This happens when the user does want to specify an instance. 

See line 1401: `instances = None`

`clusters`, on the other hand, would never be `None` because:
1. If the user doesn't specify a `cluster`, we will list all the clusters in line 1392.
2. If the user does specify a `cluster`, there will be at least one because we are filtering specifically for that `cluster`.

This answers half of @stug 's question. In terms of the instance logic, I think it's plausible to remove it and just use instances itself as a boolean to check whether or not we need to strip the instance header. I guess I like to leave it in for readability.